File: backend/src/main/java/com/devops/incident/model/Investigation.java
package com.devops.incident.model;

import com.devops.incident.model.enums.InvestigationStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Investigation {
    private String id;
    private String incidentId;
    private InvestigationStatus status;
    private String summary;
    private String rootCause;
    private Double confidence;
    private String aiModelUsed;
    private Instant startedAt;
    private Instant completedAt;
    private Instant createdAt;
    private Instant updatedAt;
}

File: frontend/src/types/investigation.ts
import type { Action } from './action';
import type { Evidence } from './evidence';

export type InvestigationStatus = 'PENDING' | 'IN_PROGRESS' | 'COMPLETED' | 'FAILED';

export interface Investigation {
  id: string;
  incidentId: string;
  status: InvestigationStatus;
  summary: string | null;
  rootCause: string | null;
  confidence: number | null;
  aiModelUsed: string | null;
  evidence?: Evidence[];
  actions?: Action[];
  startedAt: string | null;
  completedAt: string | null;
  createdAt: string;
  updatedAt: string;
}
